### PR TITLE
 Intersection of two Unsorted arrays in Java

### PR DESCRIPTION
**Pull request by Amanduttbhagat**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

  - Intersection : 
       We have to find intersection of two arrays :
       The intersection of two arrays is a list of numbers which are present in both the arrays
     Example :
       input : Enter the number of elements you want in the first array : 4
	           Enter the elements in the 1st array : 1 4 3 2
	 		   Enter the number of elements you want in the second array : 3
	 		   Enter the elements in the 2nd array : 1 2 3
      output : The intersection of two arrays is : 1 2 3
               Intersection of both arrays is [1 2 3] as these elements are present in both the arrays

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#6070

### CODE REVIEW

**Comment by Amanduttbhagat** [comment]
Hello @HarshCasper @ankitaggarwal23 ,
Can you review this PR

Thank you

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Use proper code Format

**Comment by Amanduttbhagat** [comment]
@ankitaggarwal23 Can you please review it again

Thank you

File: Java/cp/Intersection_of_Two_Arrays.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  - Intersection :
+       We have to find intersection of two arrays :
+       The intersection of two arrays is a list of numbers which are present in both the arrays
+     Example :
+       input : Enter the number of elements you want in the first array : 4
+	           Enter the elements in the 1st array : 1 4 3 2
+	 		   Enter the number of elements you want in the second array : 3
+	 		   Enter the elements in the 2nd array : 1 2 3
+      output : The intersection of two arrays is : 1 2 3
+               Intersection of both arrays is [1 2 3] as these elements are present in both the arrays
```
Comment (ankitaggarwal23) [inline Java/cp/Intersection_of_Two_Arrays.java]: Remove this and,
- Add Multi-Line Comments 
- Add a beginner-friendly description at the top

File: Java/cp/Intersection_of_Two_Arrays.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		}
+
+		//Creating object
+		Intersection_of_Two_Arrays obj =new Intersection_of_Two_Arrays();
+
+		System.out.print("The intersection of two arrays is : ");
+		//calling intersection method
+		obj.intersection(array1, array2);
+	}
+}
```
Comment (ankitaggarwal23) [inline Java/cp/Intersection_of_Two_Arrays.java]: rewrite it like that,
```
import java.util.Arrays;
import java.util.Scanner;

public class Intersection_of_Two_Arrays {

	//method intersection
	public void intersection(int[] array1, int[] array2) {

		int x = 0, y = 0;

		Arrays.sort(array1);
		Arrays.sort(array2);

		for(  ; x < array1.length && y < array2.length ;  ) 
		{

			if(array1[x] == array2[y]) 
			{
				System.out.print(array2[y]+" ");
				y = y+1;
				x = x+1;
			}
			else if(array1[x] < array2[y]) 
			{
				x = x+1;
			}
			else if(array1[x] > array2[y]) 
			{
				y = y+1;
			}
		}
	}

	//Driver Code
	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		System.out.print("Enter the number of elements you want in the first array : ");
		int first = sc.nextInt();

		//Creating array
		int[] array1 = new int[first];

		System.out.print("Enter the elements in the 1st array : ");
		for(int i = 0 ; i < first ; i++) {
			array1[i] = sc.nextInt();
		}

		System.out.print("Enter the number of elements you want in the second array : ");
		int second = sc.nextInt();

		//Creating array
		int[] array2 = new int[second];

		//Putting elements in the array
		System.out.print("Enter the elements in the 2nd array : ");
		for(int i = 0 ; i < second ; i++) {
			array2[i] = sc.nextInt();
		}

		//Creating object
		Intersection_of_Two_Arrays obj = new Intersection_of_Two_Arrays();

		System.out.print("The intersection of two arrays is : ");
		
		//calling intersection method
		obj.intersection(array1, array2);
	}
}
```

File: Java/cp/Intersection_of_Two_Arrays.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	 		   Enter the elements in the 2nd array : 1 2 3
+      output : The intersection of two arrays is : 1 2 3
+
+- Complexity Analysis :
+ * Time Complexity : O(n)
+         Where, n = array1.length + array2.length
+  	     We are using only one for loop, so it iterates maximum 'n' times while traversing for loop
+ * Space Complexity : O(n)
+	  	 Auxiliary space : O(1)  +  input space : O(n)
+ */
```
Comment (ankitaggarwal23) [inline Java/cp/Intersection_of_Two_Arrays.java]: rewrite it,


```
/*

Input:
Enter the number of elements you want in the first array : 4                                                                                  
Enter the elements in the 1st array : 1 2 3 4                                                                                                 
Enter the number of elements you want in the second array : 3                                                                                 
Enter the elements in the 2nd array : 1 2 3

Output:                                                                                                   
The intersection of two arrays is : 1 2 3

Time Complexity : O(n)
Space Complexity : O(n)

*/
```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Next time do like that only

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**